File: ValidAnagram.java
//https://leetcode.com/problems/valid-anagram/

class Solution {
    public boolean isAnagram(String s, String t) {
        // if(s.length() != t.length()){
        // return false;
        // }

        // Map<Character, Integer> map = new HashMap<>();
        // for(int i = 0; i < s.length(); i++){
        // map.put(s.charAt(i), map.getOrDefault(s.charAt(i), 0)+1);
        // }

        // for(int i = 0; i < t.length(); i++){
        // map.put(t.charAt(i), map.getOrDefault(t.charAt(i), 0)-1);
        // }

        // for(int val : map.values()){
        // if(val != 0){
        // return false;
        // }
        // }

        // return true;

        char[] arr1 = s.toCharArray();
        char[] arr2 = t.toCharArray();

        Arrays.sort(arr1);
        Arrays.sort(arr2);

        return (Arrays.equals(arr1, arr2));
    }
}

File: Easy/ValidAnagram/ValidAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # if len(s)!=len(t):
        #     return False
        
        # map = {}

        # for i in range (0,len(s)):
        #     map[s[i]] = map.get(s[i],0)+1
        
        # for i in range (0,len(t)):
        #     if map.get(t[i]) is None or map.get(t[i])==0:
        #         return False
        #     else:
        #         map[t[i]] = map.get(t[i])-1

        # return True

        map = [0]*26

        for i in range(0,len(s)):
            map[ord(s[i]) - ord('a')] += 1

        for i in range(0,len(t)):
            map[ord(t[i]) - ord('a')]-=1

        for x in map:
            if x!=0:
                return False
        
        return True